nexus/db-model/src/dataset.rs

```diff
@@ -49,6 +50,7 @@ impl Dataset {
         pool_id: Uuid,
         addr: Option<SocketAddrV6>,
         kind: DatasetKind,
+        zone_name: Option<String>,
```

Comment:
In the interest of making illegal states unrepresentable, I'd consider passing in the shared `DatasetKind` as an argument and then resolving it to the separate fields within here.

Reply:
Great suggestion, done in 7156ba06870c10ea01df546f10d9f7b40f07a5a1